test/parallel/test-process-should-abort-on-uncaught-setflagsfromstring.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const v8 = require('v8');
+
+assert.strictEqual(process.shouldAbortOnUncaughtException, true);
+
+v8.setFlagsFromString('--abort-on-uncaught-exception');
+// This should make the process not crash even though the flag was passed.
+process.shouldAbortOnUncaughtException = false;
+process.once('uncaughtException', common.mustCall());
```

Comment:
My question would be whether this violates user expectations. It *might* be worthwhile emitting a process warning here so that, at the very least, the user can be aware of what is happening. It is entirely possible and likely that some userland module deep in the dependency tree could set this property and a user would have no idea why `--abort-on-uncaught-exception` is not working.

Reply:
I’m -1 on a process warning; if there is a `shouldAbortOnUncaughtException = false` statement, that’s not an accident.

> It is entirely possible and likely that some userland module deep in the dependency tree could set this property

This flag isn’t really relevant for non-diagnostics tooling, so I would be surprised if that were true.

Reply:
> ...so I would be surprised if that were true

I've been surprised many times by what I've found in userland ;-) 

If we're not going to warn on this, then the documentation for the `--abort-on-uncaught-exception` flag should be updated to clearly indicate that the argument can be short circuited by this new `process` property. If nothing else, that gives users something to go look for before filing the bug report here that the flag isn't working ;-)

Reply:
Yeah, sure. 👍